### PR TITLE
fix integrations

### PR DESCRIPTION
**Pull request by ShinnosukeUesaka**
Need further work on LLM training/evaluation.

To test, run the code in the order.
`python main.py --config-name=mnist_baseline`
`python main.py --config-name=mnist_prune`

### CODE REVIEW

File: configs/base_config.yaml
Comment (mukobi) [inline configs/base_config.yaml:9]: What's this for? Could you make the name more specific and add a comment about what it's supposed to do?

File: main.py
Comment (mukobi) [inline main.py:89]: Can you log stuff in categories using slashes? E.g. `"fine_tune/val_loss"`? Also, in practice, we'll instead probably log continuously within the fine-tuning function so we can monitor FT loss while it's going. I'd just remove this line.

File: main.py
Comment (mukobi) [inline main.py:101]: We probably want to keep the base model for the unadapt method so we can do things like a KL divergence penalty.

File: main.py
Comment (mukobi) [inline main.py:149]: Same note about logging things in categories with slashes. That way, they show up in folders in wandb.

File: src/ufm/countermeasures.py
Comment (mukobi) [inline src/ufm/countermeasures.py:20]: Why this change? I strongly think we should have the params in a config loaded by hydra.

**Review by mukobi — CHANGES_REQUESTED** [review]
For feedback, this is a big PR touching many things, and the name "fix integrations" isn't very descriptive. I'd prefer you have split this kind of things into multiple PRs in the future.

**Comment by mukobi** [comment]
Wait on #17 to merge

File: main.py
Comment (ShinnosukeUesaka) [reply main.py:101]: `ufm_model = copy.deepcopy(base_model)`
I am copying the model in line 73, and using that model to test unadapt methods.
Currently main.py finetunes in both the base model and unadapt model in one execution.
I think it would be better to instead just do one as you mentioned in the doc.

> The first time we run our script without an unadaptability method specified, it should benchmark and save the results to disk from benchmarking and fine-tuning the base model for comparison.

Comment (mukobi) [reply main.py:101]: Delete both this `del` line and the deepcopy. We should pass the base model into the unadapt method and let the individual unadapt method decide if it wants to copy the base mode before returning just the unadapted model (as is shown in the latest main.py).

File: src/ufm/countermeasures.py
Comment (ShinnosukeUesaka) [reply src/ufm/countermeasures.py:20]: I wanted to decouple the code, but it might be simpler to pass the config itself.
I will revert the change.

**Comment by ShinnosukeUesaka** [comment]
@mukobi 
The code is fully working for mnist and closely follows your main.py.

To test, please run the code in the order.
```
python main.py --config-name=mnist_baseline
python main.py --config-name=mnist_prune
```

The PR became a bit too large, I can split this into multiple PRs if you would like.

**Comment by ShinnosukeUesaka** [comment]
@mukobi 
I saw your message on Slack. I can close this pull request, because this is was intended to support both MNIST and LLMs (which added complexity as you mentioned).

I can also modify and simplify the code so that it only supports LLM if necessary. Otherwise I can just focus on countermeasures.

**Comment by mukobi** [comment]
Feel free to resolve conflicts, removing MNIST support then you can merge this in if there's significant new stuff that we'll need for LLMs. If it's only really to support MNIST models, then you can just close.

**Pull request closed without merging**
